pymc4/_model.py

```diff
@@ -83,8 +83,14 @@ def _evaluate(self):
     def make_log_prob_function(self):
         """Return the log probability of the model."""
 
-        def log_prob(*args):
-            context = contexts.InferenceContext(args, expected_vars=self._forward_context.vars)
+        def log_prob(**kwargs):
+            vars = self._forward_context.vars
+            if set(self._observations.keys()).intersection(set(kwargs.keys())):
+                raise ValueError('double specified var')
+            kwargs.update(self._observations)
+            if len(kwargs) != len(vars):
```

Comment:
Out of scope for this PR but adding note. Having `pm.RandomVariable` objects referenced by key in a dictionary would help with checks in the `log_prob` function

https://github.com/pymc-devs/pymc4/issues/112